core/src/test/java/nl/nn/adapterframework/jdbc/JdbcTestBase.java

```diff
@@ -52,12 +56,25 @@
 		return dataSourceFactory.getAvailableDataSources();
 	}
 
+	public enum TransactionManagerType {
+		DATASOURCE, BTM, NARAYANA
+	}
+	
 	@Before
 	public void setup() throws Exception {
-		if(dataSource instanceof DriverManagerDataSource) {
-			Properties dataSourceProperties = ((DriverManagerDataSource)dataSource).getConnectionProperties();
-			productKey = dataSourceProperties.getProperty(URLDataSourceFactory.PRODUCT_KEY);
-			testPeekShouldSkipRecordsAlreadyLocked = Boolean.parseBoolean(dataSourceProperties.getProperty(URLDataSourceFactory.TEST_PEEK_KEY));
+		switch (transactionManagerType) {
+			case DATASOURCE:
+				Properties dataSourceProperties = ((DriverManagerDataSource)dataSource).getConnectionProperties();
+				productKey = dataSourceProperties.getProperty(URLDataSourceFactory.PRODUCT_KEY);
+				testPeekShouldSkipRecordsAlreadyLocked = Boolean.parseBoolean(dataSourceProperties.getProperty(URLDataSourceFactory.TEST_PEEK_KEY));
+				break;
+			case BTM:
+				productKey = ((PoolingDataSource)dataSource).getUniqueName();
+				break;
+	//		case NARAYANA:
+	//			return new NarayanaXADataSourceFactory();
```

Comment:
Moet dit? Kan deze ook aan? Of is dit nog niet geimplementeerd? (dan NotImplementedException)

Reply:
fixed